src/osal/linux/osal.c

```diff
@@ -37,7 +37,7 @@
 #include <time.h>
 #include <unistd.h>
 
-#define PF_FILE_VERSION    1
+#define PF_FILE_VERSION    1        /** Increase when any of the structs we store in file is modified. Max 999. */
```

Comment:
Wouldn't the magic/version stuff be the same for all platforms? Why not compute and check it in the stack? It could be supplied to these functions as a blob to prepend the useful data. A simple hex value would probably be enough (say 32-bits for the magic and 32-bits for the version at the start of the file).

Reply:
Moved into the stack. Using two uint32_t in the start of the file.